lib/dependency-manager-adapters/npm.js

```diff
@@ -53,16 +77,22 @@ module.exports = CoreObject.extend({
     });
   },
   cleanup: function() {
-    var adapter = this;
-    return adapter._restoreOriginalDependencies().then(function() {
+    return this._restoreOriginalDependencies().then(() => {
       debug('Remove backup package.json and node_modules');
-      return RSVP.all([rimraf(path.join(adapter.cwd, adapter.packageJSONBackupFileName)),
-        rimraf(path.join(adapter.cwd, adapter.nodeModulesBackupLocation))]);
+
+      let cleanupTasks = [rimraf(path.join(this.cwd, this.packageJSONBackupFileName)),
+        rimraf(path.join(this.cwd, this.nodeModulesBackupLocation))];
+
+      if (this.yarnLockPresent) {
+        cleanupTasks.push(rimraf(path.join(this.cwd, this.yarnLockBackupFileName)));
+      }
+
+      return RSVP.all(cleanupTasks);
     }).catch(function(e) {
       console.log('Error cleaning up npm scenario:', e);
     })
-    .then(function() {
-      return adapter._install();
+    .then(() => {
```

Comment:
@kategengler I am not really that familiar with `Eslint`. Maybe we can force `0.12` syntax with it?